address_multi.go

```diff
@@ -0,0 +1,195 @@
+package iotago
+
+import (
+	"bytes"
+	"context"
+
+	"golang.org/x/crypto/blake2b"
+
+	"github.com/iotaledger/hive.go/ierrors"
+	"github.com/iotaledger/hive.go/lo"
+	"github.com/iotaledger/hive.go/serializer/v2"
+	"github.com/iotaledger/iota.go/v4/hexutil"
+)
+
+const (
+	AddressWeightSerializedBytesSize = serializer.OneByte
+)
+
+var (
+	ErrMultiAddrCannotBeReconstructedViaBech32 = ierrors.New("multi address cannot be reconstructed via bech32")
+	ErrMultiAddressThresholdInvalid            = ierrors.New("multi address treshold invalid")
+)
+
+// AddressWithWeight is an Address with a weight used for threshold calculation in a MultiAddress.
+type AddressWithWeight struct {
+	Address Address `serix:"0,mapKey=address"`
+	Weight  byte    `serix:"1,mapKey=weight"`
+}
+
+func (a *AddressWithWeight) Size() int {
+	// address + weight
+	return a.Address.Size() + AddressWeightSerializedBytesSize
+}
+
+// AddressesWithWeight is a set of AddressWithWeight.
+type AddressesWithWeight []*AddressWithWeight
+
+// MultiAddress defines a multi address that consists of addresses with weights and
+// a threshold value that needs to be reached to unlock the multi address.
+// It has a capability bitmask that enables additional features.
+type MultiAddress struct {
+	Addresses           AddressesWithWeight        `serix:"0,mapKey=addresses"`
+	Threshold           uint16                     `serix:"1,mapKey=threshold"`
+	AllowedCapabilities AddressCapabilitiesBitMask `serix:"2,mapKey=allowedCapabilities,lengthPrefixType=uint8,maxLen=1"`
+}
+
+func (addr *MultiAddress) Clone() Address {
+	cpy := &MultiAddress{
+		Addresses: make(AddressesWithWeight, 0, len(addr.Addresses)),
+		Threshold: addr.Threshold,
```

Comment:
this is not copying the `AllowedCapabilities`, so either remove that all from this PR in favor of the other one or fix it in this